source/common/http/http1/codec_impl.cc

```diff
@@ -742,7 +695,7 @@ Envoy::StatusOr<ConnectionImpl::HttpParserCode> ConnectionImpl::onHeadersComplet
       handling_upgrade_ = true;
     }
   }
-  if (parser_.method == HTTP_CONNECT) {
+  if (parser_->methodName() == Headers::get().MethodValues.Connect) {
```

Comment:
minor perf/readability nit: capture out Headers::get() as a temp, here and elsewhere in this file.